myst_nbs/gaussian_processes/MOGP-Coregion-Hadamard.myst.md

```diff
@@ -0,0 +1,355 @@
+
+The LCM is a generalization of the ICM with two or more input kernels, so the LCM kernel is basically a sum of several ICM kernels. The LMC allows several independent samples from GPs with different covariances (kernels).
+
+In this example, in addition to an `ExpQuad` kernel, we add a `Matern32` kernel for input data.
+
+$$ K_{LCM} = B \otimes K_{ExpQuad} + B \otimes K_{Matern32} $$
+
+```{code-cell} ipython3
+def get_lcm(input_dim, active_dims, num_outputs, kernels, W=None, B=None, name="ICM"):
+    """
+    This function generates a LCM kernel from a list of input `kernels` and a Coregion kernel.
+    """
+    if B is None:
+        kappa = pm.Gamma(f"{name}_kappa", alpha=5, beta=1, shape=num_outputs)
+        if W is None:
+            W = pm.Normal(
+                f"{name}_W",
+                mu=0,
+                sigma=5,
+                shape=(num_outputs, 1),
+                initval=np.random.randn(num_outputs, 1),
+            )
+    else:
+        kappa = None
+
+    cov_func = 0
+    for idx, kernel in enumerate(kernels):
+        icm = get_icm(input_dim, kernel, W, kappa, B, active_dims)
+        cov_func += icm
+    return cov_func
+```
+
+```{code-cell} ipython3
+with pm.Model() as model:
+    # Priors
+    ell = pm.Gamma("ell", alpha=2, beta=0.5, shape=2)
+    eta = pm.Gamma("eta", alpha=3, beta=1, shape=2)
+    kernels = [pm.gp.cov.ExpQuad, pm.gp.cov.Matern32]
+    sigma = pm.HalfNormal("sigma", sigma=3)
+
+    # Define a list of covariance functions
+    cov_list = [
+        eta[idx] ** 2 * kernel(input_dim=2, ls=ell[idx], active_dims=[0])
+        for idx, kernel in enumerate(kernels)
+    ]
+
+    # Get the LCM kernel
+    cov_lcm = get_lcm(input_dim=2, active_dims=[1], num_outputs=n_outputs, kernels=cov_list)
+
+    # Define a Multi-output GP
+    mogp = pm.gp.Marginal(cov_func=cov_lcm)
+    y_ = mogp.marginal_likelihood("f", X, Y, sigma=sigma)
+```
+
+```{code-cell} ipython3
+pm.model_to_graphviz(model)
+```
+
+```{code-cell} ipython3
+%%time
+with model:
+    gp_trace = pm.sample(2000, chains=1)
+```
+
+### Prediction
+
+```{code-cell} ipython3
+:tags: []
+
+%%time
+with model:
+    preds = mogp.conditional("preds", X_new)
+    gp_samples = pm.sample_posterior_predictive(gp_trace, var_names=["preds"], random_seed=42)
+```
+
+```{code-cell} ipython3
+plot_predictive_posteriors(f_pred, top_pitchers, M, X_new)
+```
+
+```{code-cell} ipython3
+az.plot_trace(gp_trace)
+plt.tight_layout()
+```
+
+## Acknowledgement
+This work is supported by 2022 [Google Summer of Codes](https://summerofcode.withgoogle.com/) and [NUMFOCUS](https://numfocus.org/).
+
++++
+
+## Authors
+* Authored by [Danh Phan](https://github.com/danhphan), [Bill Engels](https://github.com/bwengals), [Chris Fonnesbeck](https://github.com/fonnesbeck) in November, 2022 ([pymc-examples#454](https://github.com/pymc-devs/pymc-examples/pull/454))
+
++++
+
+## References
+:::{bibliography}
+:filter: docname in docnames
+:::
+
```

Comment:
Right before this there should be the watermark title: https://docs.pymc.io/en/latest/contributing/jupyter_style.html#watermark